### ISSUE TITLE
[Bug]: ActionMenu items can't be selected with keyboard after selecting item with mouse

### ISSUE THREAD

**Issue by audknu** [adobe/spectrum-web-components#4132]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-action-menu

### Expected behavior

Regardless of how the user was previously interacting with the Action Menu (e.g. w/ a mouse), when the user navigates to the ActionMenu button and enters into the menu with the keyboard, they should able to select an item in the menu by hitting "return" on the keyboard.

### Actual behavior

After a user clicks on an item in the Action Menu using their mouse, the user is unable to select an item in the menu with their keyboard.

### Screenshots

When the user tries to select an item in the menu with the keyboard after previously selecting an item with their mouse:

https://github.com/adobe/spectrum-web-components/assets/24359834/6cbfdd0a-51b5-4e1d-b5e4-22e6da75b543

When a user tries to select an item with only keyboard (does not interact with Action Menu with the mouse):

https://github.com/adobe/spectrum-web-components/assets/24359834/99ddace2-4e96-4942-ad89-2d85a418b840

### What browsers are you seeing the problem in?

Chrome

### How can we reproduce this issue?

1. Go to an ActionMenu example (e.g., [SWC doc](https://opensource.adobe.com/spectrum-web-components/components/action-menu/#sizes))
2. Click on Action Menu button **with your mouse**
3. Click on an item in the menu **with your mouse**
4. Now navigate back to the menu with your keyboard
5. Try to select item with your keyboard
6. Observe hitting return on the keyboard does not select the item
